app/javascript/tests/layout/Header.test.js

```diff
@@ -0,0 +1,183 @@
+import React from 'react';
+import _ from 'lodash';
+import { shallow } from 'enzyme';
+import { Nav, Navbar } from 'react-bootstrap';
+import Header from '../../components/layout/Header';
+import { mockUser1, mockUser2 } from '../mocks/mockUsers';
+
+const helpLinks = {
+  user_guides: 'https://www.sara.org/user-guides',
+  user_forum: 'https://www.sara.org/user-forum',
+  contact_us: 'https://www.sara.org/contact-us',
+};
+const version = '123';
+
+/**
+ * If additionalProps undefiend, the Header component is rendered with the default props included below
+ * If any props are defined in additional props, they overwrite whatever the default value was set as
+ */
+function getWrapper(additionalProps) {
+  return shallow(<Header report_mode={false} version={version} show_demo_warning_background={false} banner_message="" current_user={mockUser1} help_links={helpLinks} {...additionalProps} />);
+}
+
+describe('Header', () => {
+  it('Properly renders all main components', () => {
+    const wrapper = getWrapper({ current_user: mockUser1 });
+    const primaryNav = wrapper.find(Navbar).find('.primary-nav');
+    expect(primaryNav.exists()).toBe(true);
+    expect(primaryNav.find(Navbar.Brand).exists()).toBe(true);
+    expect(primaryNav.find(Navbar.Brand).text()).toContain(`Sara Alert${version}`);
+    expect(primaryNav.find(Navbar.Text).exists()).toBe(true);
+    expect(primaryNav.find(Navbar.Text).text()).toContain(`${mockUser1.email} (${_.startCase(mockUser1.role)})`);
+    expect(
+      primaryNav
+        .find(Nav.Link)
+        .find({ href: `${window.BASE_PATH}/users/sign_out` })
+        .exists()
+    ).toBe(true);
+    expect(
+      primaryNav
+        .find(Nav.Link)
+        .find({ href: `${window.BASE_PATH}/users/sign_out` })
+        .text()
+    ).toEqual('Logout');
+    expect(primaryNav.find(Nav.Link).find({ id: 'helpMenuButton' }).exists()).toBe(true);
+    expect(primaryNav.find(Nav.Link).children().find('i').exists()).toBe(true);
+    expect(primaryNav.find('.dropdown-menu').exists()).toBe(true);
+    expect(primaryNav.find('.enroller-tab').exists()).toBe(false);
+    expect(primaryNav.find('.monitoring-tab').exists()).toBe(false);
+    expect(primaryNav.find('.admin-panel-tab').exists()).toBe(false);
+    expect(primaryNav.find('.analytics-tab').exists()).toBe(false);
+    expect(primaryNav.find('.api-tab').exists()).toBe(false);
+    expect(primaryNav.find('.jobs-tab').exists()).toBe(false);
+    expect(wrapper.find(Navbar).find('.banner-message').exists()).toBe(false);
+  });
+
+  it('Shows "Enroller Dashboard" tab when user. can_see_enroller_dashboard_tab', () => {
```

Comment:
is there intentionally a space here?  makes it seem like it's the end of a sentence.  I'd either remove the period or remove the space (same with the rest of these tests)

Reply:
I'm just noticing that this is what I suggested... must have been a typo sorry!

Reply:
[4eacbdb](https://github.com/SaraAlert/SaraAlert/pull/1243/commits/4eacbdb31f43a07a69fb35000a7ecb3ac56ab60e)